src/drivers/apl-ssp.c

```diff
@@ -242,18 +243,26 @@ static inline int ssp_set_config(struct dai *dai,
 		inverted_frame = true; /* handled later with format */
 		break;
 	case SOF_DAI_FMT_IB_IF:
-		sspsp |= SSPSP_SCMODE(2);
+		inverted_clock = true; /* handled later with bclk idle */
 		inverted_frame = true; /* handled later with format */
 		break;
 	case SOF_DAI_FMT_IB_NF:
-		sspsp |= SSPSP_SCMODE(2);
+		inverted_clock = true; /* handled later with bclk idle */
 		break;
 	default:
 		trace_ssp_error("ec3");
 		ret = -EINVAL;
 		goto out;
 	}
 
+	/* supporting bclk idle state */
+	if (ssp->params.clks_control &
+		SOF_DAI_INTEL_SSP_CLKCTRL_BCLK_IDLE) {
+		sspsp |= SSPSP_SCMODE((inverted_clock ^ 0x3) & 0x3);
```

Comment:
is the ^ 0x3 correct? You are only looking at one bit here. or I am missing the optimization.

Reply:
There is no bit directly responsible for idle state. Clock signal polarity and clock idle state are interrelated. SCMODE in SSPSP can be programmed:
00: Data driven (falling), data sampled (rising), idle state (low)
01: Data driven (rising), data sampled (falling), idle state (low)
10: Data driven (rising), data sampled (falling), idle state (high)
11: Data driven (falling), data sampled (rising), idle state (high)